src/main/java/de/zalando/aruha/nakadi/controller/TopicsController.java

```diff
@@ -72,8 +42,8 @@
     public ResponseEntity<?> listPartitions(@PathVariable("topicId") final String topicId) {
         try {
             return ok().body(topicRepository.listPartitions(topicId));
-        } catch (final NakadiException e) {
-            return status(503).body(e.getProblemMessage());
+        } catch (final Exception e) {
```

Comment:
This is not correct. NakadiException (or the ones that are known in a controlled manner) can be a source of 503, but not the general case like this. The catch all should be a 500, but 503 is a specific case.